lib/ansible/playbook/role_include.py

```diff
@@ -43,6 +43,11 @@ class IncludeRole(TaskInclude):
     circumstances related to the `- include_role: ...`
     """
 
+    BASE = set(['name', 'role'])
+    FROM_ARGS = set(['tasks_from', 'vars_from', 'defaults_from'])
+    OTHER_ARGS = set(['private', 'allow_duplicates'])
+    VALID_ARGS = frozenset(BASE.union(FROM_ARGS.union(OTHER_ARGS)))
```

Comment:
Nonblocking/doesn't matter:  I like to use frozenset and tuples for immutables.  Since these sets are never updated you can do:
``` python
    BASE = frozenset(('name', 'role'))
    FROM_ARGS = frozenset(('tasks_from', 'vars_from', 'defaults_from'))
    OTHER_ARGS = frozenset(('private', 'allow_duplicates'))
    VALID_ARGS = frozenset(BASE.union(FROM_ARGS.union(OTHER_ARGS)))